packages/tooltip/src/vaadin-tooltip.js

```diff
@@ -357,8 +362,14 @@ class Tooltip extends ThemePropertyMixin(ElementMixin(PolymerElement)) {
   }
 
   /** @private */
-  __onFocusout() {
+  __onFocusout(event) {
+    // Do not close when moving focus within a component.
+    if (this.target.contains(event.relatedTarget)) {
+      return;
+    }
+
     this.__focusInside = false;
+    this.__escPressed = false;
```

Comment:
nit: not tested

Reply:
Removed the flag and changed `focusin` listener to also check for `event.relatedTarget`.

Reply:
The new `&& this.target.contains(event.relatedTarget)` condition still doesn't seem to be tested (all tests pass with the condition removed)

Reply:
Fixed the logic and added tests to also ensure that tooltip doesn't get re-opened after mousedown.

Reply:
The `this.target.contains(event.target) && ` can still be removed from the check inside `__onFocusIn` because the condition should always be true